### DISCUSSION TITLE
Feature request: normalising Github topics and showing count

### DISCUSSION THREAD

**Discussion by karlicoss** [community/community#22987] [New to GitHub]
<p>First, thanks to Github developers for topics, they are a great way of discovering projects and refining search!</p>
<p>There is one thing I lack about them which is some sort of topic normalisation/unification.</p>
<p>Some topics are multiword and I always strugle when choosing which topic to use in such case. E.g. imagine you implement a bot for reddit: you could either reddit-bot or redditbot tags.</p>
<p><img src="https://user-images.githubusercontent.com/291333/181074134-4d651965-a061-4629-be8d-fdc1d864ae69.jpeg" alt="topics-current.jpg" data-base62-sha1="mdKugeMBbbstNtTqW3n6j0mtwWg" width="438" height="398"></p>
<p>And indeed, if you search for them, both give different results:</p>
<ul>
<li>
<a href="https://github.com/search?q=reddit-bot&amp;type=Topics" rel="nofollow noopener">reddit-bot</a> : 296 results</li>
<li>
<a href="https://github.com/search?q=redditbot&amp;type=Topics" rel="nofollow noopener">redditbot</a>:  22 results</li>
</ul>
<p>My suggestions would be</p>
<ul>
<li>change topic search algorighm so hyphens/underscores (or perhaps any special characters) are ignored. Worst case it would add few false positives (although I’d struggle to come up with example)</li>
<li>suggest topic statistics when you add topics to your repositories. Then perhaps ambigous tags would happen less because the most common topic (reddit-bot in this case) would be suggested first. Here’s how Stackexchange does it for inspiration:</li>
</ul>
<p><img src="https://user-images.githubusercontent.com/291333/181074135-412fd5f1-a129-41d0-ae90-06079f635036.jpeg" alt="tags-suggest.jpg" data-base62-sha1="vkyDtJ2rP0cmTZUGChbCaaLTnAk" width="675" height="384"></p>
<p>Thanks!</p>

**Comment by MChevy422** [comment, marked as answer]
<p>Hello <span><a class="mention" href="/karlicoss">@karlicoss</a></span>, </p>
<p>Thank you for your feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.<br><br>Once again, thank you for your input!<br><br>Greatly appreciated,<br><br>MChevy422</p>
